src/plugin/parse-html/src/java/org/apache/nutch/parse/html/DOMContentUtils.java

```diff
@@ -428,6 +438,15 @@ public void getOutlinks(URL base, ArrayList<Outlink> outlinks, Node node) {
                 URL url = URLUtil.resolveURL(base, target);
                 outlinks.add(new Outlink(url.toString(), linkText.toString()
                     .trim()));
+										
+                // NUTCH-2433 - Keep the node name where the URL was found into 
+                // the outlink metadata
+                if (keepNodenames) {
+                  MapWritable metadata = new MapWritable();
+                  metadata.put(new Text(srcTagMetaName), new Text(nodeName));
+                  outlinks.get(outlinks.size() - 1).setMetadata(metadata);
```

Comment:
Maybe keep a reference to the last outlink above to avoid the extra get operation. Only a matter of taste, of course.

Reply:
Good point, I agree. I've updated the code accordingly.